javaslang/src/main/java/javaslang/collection/BitMappedTrie.java

```diff
@@ -55,10 +55,29 @@ private static int treeSize(int branchCount, int depthShift) {
         return branchCount * fullBranchSize;
     }
 
-    BitMappedTrie<T> prepend(T leading) {
-        final int newSize = length() + 1;
+    static <T> BitMappedTrie<T> ofAll(Iterable<? extends T> iterable, int size) {
+        if (size == 0) { return empty(); }
+
+        Object[] array = (iterable instanceof Collection<?>) ? ((Collection<?>) iterable).toArray()
```

Comment:
Let's call `BitMappedTrie.ofAll(veryLongCollection)`.
Here is the first copy loop, and in `Arrays.grouped()` is second one, without any changes of array between them.